pynpoint/processing/psfpreparation.py

```diff
@@ -104,9 +102,19 @@ def run(self) -> None:
 
         # Get the number of images and split into batches to comply with memory constraints
         im_shape = self.m_image_in_port.get_shape()
-        nimages = im_shape[0]
+        nimages = im_shape[-3]
```

Comment:
Add something like `ndim = self.m_image_in_port.get_ndim()` here and use the value of `ndim` for the `if` conditions where the number of dimensions (3 or 4) is required.

Reply:
I can do this, but the -3 basically does this in one step instead of two if conditions.

Reply:
Sorry I meant adding an extra line with `ndim =` and using the `ndim` value whenever the number of dimensions is required.

Reply:
That makes sens. I'll change it.